Add timeout configs to CloseableHttpClient to verify it we can overco…

sridharmunendramani96 opened the pull request:
Add timeout configs to CloseableHttpClient to verify it we can overcome unexpected errors during saving the test

sridharmunendramani96 committed:
Add timeout configs to CloseableHttpClient to verify it we can overcome unexpected errors during saving the test

snevesbarros commented on angles-java-core/src/main/java/com/github/angleshq/angles/api/requests/BaseRequests.java:
```diff
@@ -20,7 +21,19 @@
 public abstract class BaseRequests {
 
-    protected CloseableHttpClient client = HttpClients.createDefault();
+    protected int connectionTimeout = 5000;
```
Rather than hardcoding these timeouts, it probably extracting them into a properties file and then using maven filtering and pom properties to set defaults (that can be overridden).

snevesbarros replied:
Actually maybe they can be passed as part of the constructor (e.g. a the RequestConfig object) and then have an additional arguments that can be passed for the Angles Reporter.

sridharmunendramani96 replied:
Thanks, I have update the pull request to pass the timeouts via constructor from Angles Reporter.

sridharmunendramani96 committed:
Add timeout configs to CloseableHttpClient

Add connectionTimeouts and socketTimeouts to CloseableHttpClient during ExecutionRequests to see if we can avoid random timeouts.

snevesbarros commented on angles-java-core/src/main/java/com/github/angleshq/angles/AnglesReporter.java:
```diff
@@ -35,12 +36,20 @@ public class AnglesReporter implements AnglesReporterInterface {
 
+    protected static int connectionTimeout = 5000;
+    protected static int socketTimeout = 10000;
+    private static RequestConfig.Builder requestConfig = RequestConfig.custom();
```
This variable is not needed as you're not storing the config.

sridharmunendramani96 committed:
Update AnglesReporter.java

Remove unused requestConfig.
